src/Microsoft.OData.Core/MediaTypeUtils.cs

```diff
@@ -547,13 +549,15 @@ private static void ConvertApplicationJsonInAcceptableMediaTypes(IList<KeyValueP
                     HttpUtils.CompareMediaTypeNames(mediaType.Type, MimeConstants.MimeApplicationType))
                 {
                     if (mediaType.Parameters == null ||
-                        !mediaType.Parameters.Any(p => HttpUtils.CompareMediaTypeParameterNames(p.Key, MimeConstants.MimeMetadataParameterName)))
+                        !mediaType.Parameters.Any(p => HttpUtils.CompareMediaTypeParameterNames(p.Key, MimeConstants.MimeMetadataParameterName) || HttpUtils.CompareMediaTypeParameterNames(p.Key, MimeConstants.MimeShortMetadataParameterName)))
                     {
                         // application/json detected; convert it to Json Light
                         IList<KeyValuePair<string, string>> existingParams = mediaType.Parameters != null ? mediaType.Parameters.ToList() : null;
                         int newCount = existingParams == null ? 1 : existingParams.Count + 1;
                         List<KeyValuePair<string, string>> newParams = new List<KeyValuePair<string, string>>(newCount);
-                        newParams.Add(new KeyValuePair<string, string>(MimeConstants.MimeMetadataParameterName, MimeConstants.MimeMetadataParameterValueMinimal));
+                        newParams.Add(new KeyValuePair<string, string>(
+                            version < ODataVersion.V401 ? MimeConstants.MimeMetadataParameterName : MimeConstants.MimeShortMetadataParameterName,

```

Comment:
> < [](start = 35, length = 2)

Sorry, would you please help double check the syntax is what you are expected.

Reply:
For ODataVersions prior to 4.01, the application/json metadata type parameter should be called "odata.metadata".  For 4.01 and beyond, the header should be called "metadata".  The default value remains "minimal".

Adding some test cases and simplifying the code a bit for comparing prefixed with non-prefixed odata metadata and odata streaming type parameters.

---
In reply to: [162159043](https://github.com/OData/odata.net/pull/1038#discussion_r162159043) [](ancestors = 162159043)